src/openrct2/peep/GuestPathfinding.cpp

```diff
@@ -2287,7 +2287,8 @@ bool GuestPathfinding::IsValidPathZAndDirection(TileElement* tileElement, int32_
     }
     else
     {
-        if (currentZ != tileElement->BaseHeight)
+        if (currentZ != tileElement->BaseHeight && currentZ - 1 != tileElement->BaseHeight
+            && currentZ + 1 != tileElement->BaseHeight)
```

Comment:
I wonder whether it would be simpler to just do:
```suggestion
        if (currentZ < ( tileElement->BaseHeight - 1 ) || currentZ > ( tileElement->BaseHeight + 1 ))
```